snap/snapcraft.yaml

```diff
@@ -727,7 +727,7 @@ parts:
     after: [go-build-helper]
     plugin: make
     source: https://github.com/hashicorp/vault.git
-    source-tag: v1.5.4
+    source-tag: v1.7.1
```

Comment:
Note we update to the patch versions in compose yesterday to pin them and this one is now `1.7.2`.
Here is the list of versions used in docker: https://github.com/edgexfoundry/edgex-compose/blob/master/compose-builder/.env#L39-L46